File: Easy/643-maximum-average-subarray-i/try-11.py
/*
 * Problem #643: Maximum Average Subarray I
 * Difficulty: Easy
 * Submission: Try 11
 * status: Accepted
 * Language: python3
 * Date: 07/12/2024, 12:46:44
 * Link: https://leetcode.com/problems/maximum-average-subarray-i/
 */

class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        window=sum(nums[:k])
        mx=window
        for i in range(k,len(nums)):
            window=window+nums[i]-nums[i-k]
            if mx<window:
                mx=window
        return mx/k
        

File: Easy/121-best-time-to-buy-and-sell-stock/try-11.cpp
/*
 * Problem #121: Best Time to Buy and Sell Stock
 * Difficulty: Easy
 * Submission: Try 11
 * status: Accepted
 * Language: cpp
 * Date: 02/12/2024, 19:14:57
 * Link: https://leetcode.com/problems/best-time-to-buy-and-sell-stock/
 */

class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int s=0,m=0,i;
        for(i=1;i<prices.size();i++){
            s+=prices[i]-prices[i-1];
            if(s<0)
            s=0;
            else if(s>m)
            m=s;
        }
        return m;
    }
};
